pkg/operator/ceph/cluster/osd/health.go

```diff
@@ -191,15 +175,30 @@ func (m *OSDHealthMonitor) removeOSDDeploymentIfSafeToDestroy(outOSDid int) erro
 	return nil
 }
 
-// updateCephStorage updates the CR with deviceclass details
-func (m *OSDHealthMonitor) updateCephStatus(devices []string) {
+// updateCephStorageStatus updates the Storage details in the ceph cluster CR
+func (m *OSDHealthMonitor) updateCephStorageStatus() {
 	cephCluster := cephv1.CephCluster{}
 	cephClusterStorage := cephv1.CephStorage{}
 
+	devices, err := client.GetDeviceClasses(m.context, m.clusterInfo)
+	if err != nil {
+		logger.Errorf("failed to get osd device classes. %v", err)
+		return
+	}
+
 	for _, device := range devices {
```

Comment:
How about renaming these while we're changing this method? It's confusing as `devices`.
```suggestion
	for _, deviceClass := range deviceClasses {
```